django/contrib/admin/options.py

```diff
@@ -1814,7 +1814,12 @@ def _get_obj_does_not_exist_redirect(self, request, opts, object_id):
 
     @csrf_protect_m
     def changeform_view(self, request, object_id=None, form_url="", extra_context=None):
-        with transaction.atomic(using=router.db_for_write(self.model)):
+        db = (
+            router.db_for_read(self.model)
```

Comment:
I'm not sure of the value of creating a transaction here as Django recommends running in `READ COMMITTED` which means that creating a transaction is moot as there won't be any snapshot isolation for reads.

Reply:
It ought to be unnecessary to create a transaction - I'll remove it.

Reply:
Fixed in [fefc3b7](https://github.com/django/django/pull/18268/commits/fefc3b7e02b3da670fbc9ab8855251c4e034eb4e)